src/ServiceControl.Monitoring/Http/HttpEndpoint.cs

```diff
@@ -8,31 +8,55 @@
 
 namespace ServiceControl.Monitoring.Http
 {
+    using System;
+    using System.Configuration;
+    using System.Linq;
+
     class HttpEndpoint : Feature
     {
+        public const string AppConfigHostnameKey = "HttpEndpoint/Hostname";
+        public const string AppConfigPort = "HttpEndpoint/Port";
+
         public HttpEndpoint()
         {
             DependsOn<MetricsReceiver>();
         }
 
         protected override void Setup(FeatureConfigurationContext context)
         {
-            context.RegisterStartupTask(BuildTask);
+            var host = MetricsApiModule.DefaultHost;
+            var hostname = GetValue(AppConfigHostnameKey);
+            var port = GetValue(AppConfigPort);
+            int portValue;
+
+            if (string.IsNullOrWhiteSpace(hostname) == false &&
+                string.IsNullOrWhiteSpace(port) == false &&
+                int.TryParse(port, out portValue))
+            {
+                host = new Uri(host + ":" + port);
+            }
```

Comment:
I think it should. I mean absence of value is fine but non-integer should throw.